common/src/main/java/org/esbtools/message/admin/common/EsbMessageAdminServiceImpl.java

```diff
@@ -18,94 +18,83 @@
-import org.esbtools.message.admin.common.dao.AuditEventDAOImpl;
-import org.esbtools.message.admin.common.dao.EsbErrorDAO;
-import org.esbtools.message.admin.common.dao.EsbErrorDAOImpl;
-import org.esbtools.message.admin.common.dao.MetadataDAO;
-import org.esbtools.message.admin.common.dao.MetadataDAOImpl;
+import org.esbtools.message.admin.common.config.VisibilityConfiguration;
 import org.esbtools.message.admin.common.extractor.KeyExtractorException;
 import org.esbtools.message.admin.common.extractor.KeyExtractorUtil;
-import org.esbtools.message.admin.common.utility.EncryptionUtil;
+import org.esbtools.message.admin.common.orm.AuditEventEntity;
+import org.esbtools.message.admin.common.orm.EsbMessageEntity;
+import org.esbtools.message.admin.common.orm.EsbMessageHeaderEntity;
+import org.esbtools.message.admin.common.orm.MetadataEntity;
+import org.esbtools.message.admin.common.utility.ConversionUtility;
+import org.esbtools.message.admin.common.utility.EncryptionUtility;
+import org.esbtools.message.admin.model.AuditEvent;
+import org.esbtools.message.admin.model.Criterion;
 import org.esbtools.message.admin.model.EsbMessage;
+import org.esbtools.message.admin.model.Header;
+import org.esbtools.message.admin.model.HeaderType;
 import org.esbtools.message.admin.model.MetadataField;
 import org.esbtools.message.admin.model.MetadataResponse;
 import org.esbtools.message.admin.model.MetadataType;
 import org.esbtools.message.admin.model.SearchCriteria;
 import org.esbtools.message.admin.model.SearchResult;
-import org.json.simple.JSONObject;
-import org.json.simple.parser.JSONParser;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.persistence.EntityManager;
+import javax.persistence.NoResultException;
+import javax.persistence.Query;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.esbtools.message.admin.common.config.EMAConfiguration.getEncryptionKey;
+import static org.esbtools.message.admin.common.config.EMAConfiguration.getNonViewableMessages;
+import static org.esbtools.message.admin.common.config.EMAConfiguration.getPartiallyViewableMessages;
+import static org.esbtools.message.admin.common.config.EMAConfiguration.getResyncRestEndpoints;
+import static org.esbtools.message.admin.common.config.EMAConfiguration.getSortingFields;
+import static org.esbtools.message.admin.common.config.EMAConfiguration.getSuggestedFields;
+
 @Named
 public class EsbMessageAdminServiceImpl implements Provider {
```

Comment:
I know this is out of scope, but can we rename Provider to EsbMessageAdminService ? Never understood why "Provider"